service/api/temporal/client.go

```diff
@@ -82,16 +85,59 @@ func (t *temporalClient) DescribeWorkflowExecution(ctx context.Context, workflow
 	if err != nil {
 		return nil, err
 	}
+
+	searchAttributes, err := mapToIwfSearchAttributes(resp.GetWorkflowExecutionInfo().GetSearchAttributes())
+	if err != nil {
+		return nil, err
+	}
+
 	return &api.DescribeWorkflowExecutionResponse{
-		RunId:  resp.GetWorkflowExecutionInfo().GetExecution().GetRunId(),
-		Status: status,
+		RunId:            resp.GetWorkflowExecutionInfo().GetExecution().GetRunId(),
+		Status:           status,
+		SearchAttributes: searchAttributes,
 	}, nil
 }
 
+func mapToIwfSearchAttributes(searchAttributes *common.SearchAttributes) (map[string]iwfidl.SearchAttribute, error) {
+	result := make(map[string]iwfidl.SearchAttribute)
+	if searchAttributes == nil {
+		return result, nil
+	}
+
+	for key, value := range searchAttributes.IndexedFields {
+		var object interface{}
+		err := converter.GetDefaultDataConverter().FromPayload(value, &object)
+		if err != nil {
+			return make(map[string]iwfidl.SearchAttribute), nil
+		}
+
+		str, isString := object.(string)
+		if isString {
+			result[key] = iwfidl.SearchAttribute{
+				Key:         iwfidl.PtrString(key),
+				StringValue: &str,
+				ValueType:   iwfidl.PtrString(service.SearchAttributeValueTypeKeyword),
+			}
+		}
+		integer, isInt := object.(float64)
```

Comment:
@longquanzheng debugged and found the returned type is float64..., maybe a bug in temporal

Reply:
https://github.com/temporalio/sdk-go/issues/942 opened a bug ticket

Reply:
should we do the casting to json.Number/float64 and then get/cast to int64? or should we create a data converter?